packages/react-composites/review/react-composites.api.md

```diff
@@ -753,6 +770,62 @@ export type TopicChangedListener = (event: {
     topic: string;
 }) => void;
 
+// @beta (undocumented)
+export const UPLOAD_CANCELLED_EVENT = "uploadCancelled";
+
+// @beta (undocumented)
+export const UPLOAD_COMPLETED_EVENT = "uploadCompleted";
+
+// @beta (undocumented)
+export const UPLOAD_FAILED_EVENT = "uploadFailed";
+
+// @beta (undocumented)
+export const UPLOAD_PROGRESSED_EVENT = "uploadProgressed";
+
+// @beta
+export type UploadCanceledListener = () => void;
+
+// @beta
+export type UploadCompleteListener = (metaData: FileMetaData) => void;
+
+// @beta
+export class UploadedFile implements UploadEvents {
+    constructor(file: File);
+    cancelUpload(): void;
+    completeUpload(metaData: FileMetaData): void;
+    extension(): string;
+    failUpload(message: string): void;
+    file: File;
+    // (undocumented)
+    isUploaded(): boolean;
+    metaData?: FileMetaData;
+    off(event: UploadedFileEvents, listener: UploadedFileEventListener): void;
+    on(event: UploadedFileEvents, listener: UploadedFileEventListener): void;
+    progress: number;
+    truncatedName(length?: number): string;
```

Comment:
when we upload does the file name length matter or is it more about impacting the UI when we need to render it? I dont think we want to remove information on the upload when it makes more sense to truncate things when rendering the files if the names are too long

Reply:
This function is only used render the file name in file cards.
we are not truncating the file names for storing them in state or metadata